### PR TITLE
Change spies to conform with window.fetch API.

### PR DESCRIPTION
**Pull request by annxingyuan**
This PR addresses https://github.com/tensorflow/tfjs/issues/1140

#### Changes
- Return `Response` within a promise in stubbed fetch so it's `catch`-able
- Send content type header with fetch

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/434)

### CODE REVIEW

File: src/models_test.ts [hunk 1]
```diff
@@ -546,7 +549,11 @@ describeMathCPU('loadModel from URL', () => {
       (fileBufferMap:
            {[filename: string]: Float32Array|Int32Array|ArrayBuffer}) => {
         spyOn(window, 'fetch').and.callFake((path: string) => {
-          return new Response(fileBufferMap[path]);
+          return new Promise((res, rej) => {
```
Comment (caisq) [inline src/models_test.ts]: Google style prefers explicit, unambiguous names to short, succinct ones. So use `resolve` and `reject` instead.

Comment (caisq) [inline src/models_test.ts]: Does it make sense to remove the curly brackets in this arrow function here, since it contains only one statement?

Comment (annxingyuan) [reply src/models_test.ts]: Good point - I've removed them.

Comment (annxingyuan) [reply src/models_test.ts]: Thanks - I've updated the names.

**Pull request merged**